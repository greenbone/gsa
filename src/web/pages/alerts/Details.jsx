/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import {
  EVENT_TYPE_UPDATED_SECINFO,
  EVENT_TYPE_NEW_SECINFO,
  DELTA_TYPE_PREVIOUS,
  DELTA_TYPE_REPORT,
} from 'gmp/models/alert';
import {isDefined} from 'gmp/utils/identity';
import React from 'react';
import useTranslation from 'src/web/hooks/useTranslation';
import HorizontalSep from 'web/components/layout/HorizontalSep';
import Layout from 'web/components/layout/Layout';
import DetailsLink from 'web/components/link/DetailsLink';
import TableBody from 'web/components/table/Body';
import Col from 'web/components/table/Col';
import TableData from 'web/components/table/Data';
import InfoTable from 'web/components/table/InfoTable';
import TableRow from 'web/components/table/Row';
import Condition from 'web/pages/alerts/Condition';
import Event from 'web/pages/alerts/Event';
import Method from 'web/pages/alerts/Method';
import PropTypes from 'web/utils/PropTypes';
import withCapabilities from 'web/utils/withCapabilities';

const AlertDetails = ({capabilities, entity, reportFormats, reportConfigs}) => {
  const [_] = useTranslation();
  const {comment, condition, event, method, tasks = [], filter} = entity;
  return (
    <Layout grow flex="column">
      <InfoTable>
        <colgroup>
          <Col width="10%" />
          <Col width="90%" />
        </colgroup>
        <TableBody>
          {isDefined(comment) && (
            <TableRow>
              <TableData>{_('Comment')}</TableData>
              <TableData>{comment}</TableData>
            </TableRow>
          )}

          <TableRow>
            <TableData>{_('Condition')}</TableData>
            <TableData>
              <Condition condition={condition} event={event} />
            </TableData>
          </TableRow>

          <TableRow>
            <TableData>{_('Event')}</TableData>
            <TableData>
              <Event event={event} />
            </TableData>
          </TableRow>

          {capabilities.mayAccess('report') &&
            method?.data?.delta_type?.value === DELTA_TYPE_PREVIOUS && (
              <TableRow>
                <TableData>{_('Delta Report')}</TableData>
                <TableData>
                  {_('Previous completed report of the same task')}
                </TableData>
              </TableRow>
            )}

          {capabilities.mayAccess('report') &&
            isDefined(method?.data?.delta_type) &&
            isDefined(method?.data?.delta_report_id) &&
            method.data.delta_type.value === DELTA_TYPE_REPORT && (
              <TableRow>
                <TableData>{_('Delta Report')}</TableData>
                <TableData>
                  <span>
                    <DetailsLink
                      id={method.data.delta_report_id.value}
                      type="report"
                    >
                      {_('Report ')} {method.data.delta_report_id.value}
                    </DetailsLink>
                  </span>
                </TableData>
              </TableRow>
            )}

          <TableRow>
            <TableData>{_('Method')}</TableData>
            <TableData>
              <Method
                details={true}
                method={method}
                reportConfigs={reportConfigs}
                reportFormats={reportFormats}
              />
            </TableData>
          </TableRow>

          {isDefined(method?.data?.details_url?.value) &&
            (event.type === EVENT_TYPE_NEW_SECINFO ||
              event.type === EVENT_TYPE_UPDATED_SECINFO) && (
              <TableRow>
                <TableData>{_('Details URL')}</TableData>
                <TableData>{method.data.details_url.value}</TableData>
              </TableRow>
            )}

          {capabilities.mayAccess('filter') && isDefined(filter) && (
            <TableRow>
              <TableData>{_('Results Filter')}</TableData>
              <TableData>
                <span>
                  <DetailsLink id={filter.id} type="filter">
                    {filter.name}
                  </DetailsLink>
                </span>
              </TableData>
            </TableRow>
          )}

          <TableRow>
            <TableData>{_('Active')}</TableData>
            <TableData>{entity.isActive() ? _('Yes') : _('No')}</TableData>
          </TableRow>

          {tasks.length > 0 && (
            <TableRow>
              <TableData>{_('Tasks using this Alert')}</TableData>
              <TableData>
                <HorizontalSep $wrap>
                  {tasks.map(task => {
                    const [_] = useTranslation();

                    return (
                      <span key={task.id}>
                        <DetailsLink id={task.id} type="task">
                          {task.name}
                        </DetailsLink>
                      </span>
                    );
                  })}
                </HorizontalSep>
              </TableData>
            </TableRow>
          )}
        </TableBody>
      </InfoTable>
    </Layout>
  );
};

AlertDetails.propTypes = {
  capabilities: PropTypes.capabilities.isRequired,
  entity: PropTypes.model.isRequired,
  reportConfigs: PropTypes.array,
  reportFormats: PropTypes.array,
};

export default withCapabilities(AlertDetails);
