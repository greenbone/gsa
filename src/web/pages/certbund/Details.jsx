/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import React from 'react';
import {isDefined} from 'gmp/utils/identity';
import SeverityBar from 'web/components/bar/SeverityBar';
import Layout from 'web/components/layout/Layout';
import ExternalLink from 'web/components/link/ExternalLink';
import Col from 'web/components/table/Col';
import InfoTable from 'web/components/table/InfoTable';
import TableBody from 'web/components/table/TableBody';
import TableData from 'web/components/table/TableData';
import TableRow from 'web/components/table/TableRow';
import useTranslation from 'web/hooks/useTranslation';
import PropTypes from 'web/utils/PropTypes';

const CertBundAdvDetails = ({entity}) => {
  const [_] = useTranslation();
  const {
    title,
    version,
    severity,
    software,
    platform,
    effect,
    remoteAttack,
    risk,
    referenceSource,
    referenceUrl,
  } = entity;
  return (
    <Layout grow flex="column">
      <InfoTable>
        <colgroup>
          <Col width="10%" />
          <Col width="90%" />
        </colgroup>
        <TableBody>
          {isDefined(version) && (
            <TableRow>
              <TableData>{_('Version')}</TableData>
              <TableData>{version}</TableData>
            </TableRow>
          )}
          {isDefined(title) && (
            <TableRow>
              <TableData>{_('Title')}</TableData>
              <TableData>{title}</TableData>
            </TableRow>
          )}
          {isDefined(software) && (
            <TableRow>
              <TableData>{_('Software')}</TableData>
              <TableData>{software}</TableData>
            </TableRow>
          )}
          {isDefined(platform) && (
            <TableRow>
              <TableData>{_('Platform')}</TableData>
              <TableData>{platform}</TableData>
            </TableRow>
          )}
          {isDefined(effect) && (
            <TableRow>
              <TableData>{_('Effect')}</TableData>
              <TableData>{effect}</TableData>
            </TableRow>
          )}
          {isDefined(remoteAttack) && (
            <TableRow>
              <TableData>{_('Remote Attack')}</TableData>
              <TableData>{remoteAttack}</TableData>
            </TableRow>
          )}
          {isDefined(severity) && (
            <TableRow>
              <TableData>{_('Severity')}</TableData>
              <TableData>
                <SeverityBar severity={severity} />
              </TableData>
            </TableRow>
          )}
          {isDefined(risk) && (
            <TableRow>
              <TableData>{_('CERT-Bund Risk Rating')}</TableData>
              <TableData>{risk}</TableData>
            </TableRow>
          )}
          {isDefined(referenceSource) && (
            <TableRow>
              <TableData>{_('Reference Source')}</TableData>
              <TableData>{referenceSource}</TableData>
            </TableRow>
          )}
          {isDefined(referenceUrl) && (
            <TableRow>
              <TableData>{_('Reference URL')}</TableData>
              <TableData>
                <span>
                  <ExternalLink to={referenceUrl}>{referenceUrl}</ExternalLink>
                </span>
              </TableData>
            </TableRow>
          )}
        </TableBody>
      </InfoTable>
    </Layout>
  );
};

CertBundAdvDetails.propTypes = {
  entity: PropTypes.model.isRequired,
};

export default CertBundAdvDetails;
