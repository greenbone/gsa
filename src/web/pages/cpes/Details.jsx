/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import React from 'react';
import {isDefined} from 'gmp/utils/identity';
import SeverityBar from 'web/components/bar/SeverityBar';
import DateTime from 'web/components/date/DateTime';
import Layout from 'web/components/layout/Layout';
import DetailsLink from 'web/components/link/DetailsLink';
import Col from 'web/components/table/Col';
import InfoTable from 'web/components/table/InfoTable';
import TableBody from 'web/components/table/TableBody';
import TableData from 'web/components/table/TableData';
import TableRow from 'web/components/table/TableRow';
import useTranslation from 'web/hooks/useTranslation';
import PropTypes from 'web/utils/PropTypes';
import {renderYesNo} from 'web/utils/Render';

const CpeDetails = ({entity, links = true}) => {
  const [_] = useTranslation();
  const {title, cpeNameId, deprecated, deprecatedBy, updateTime, severity} =
    entity;
  return (
    <Layout flex="column" grow="1">
      {!isDefined(title) && (
        <p>
          {_(
            'This CPE does not appear in the CPE dictionary but is ' +
              'referenced by one or more CVE.',
          )}
        </p>
      )}

      <InfoTable>
        <colgroup>
          <Col width="10%" />
          <Col width="90%" />
        </colgroup>
        <TableBody>
          {isDefined(title) && (
            <TableRow>
              <TableData>{_('Title')}</TableData>
              <TableData>{title}</TableData>
            </TableRow>
          )}
          {isDefined(cpeNameId) && (
            <TableRow>
              <TableData>{_('CPE Name ID')}</TableData>
              <TableData>{cpeNameId}</TableData>
            </TableRow>
          )}
          <TableRow>
            <TableData>{_('Deprecated')}</TableData>
            <TableData>{renderYesNo(deprecated)}</TableData>
          </TableRow>
          {isDefined(deprecatedBy) && (
            <TableRow>
              <TableData>{_('Deprecated By')}</TableData>
              <TableData>
                <DetailsLink id={deprecatedBy} textOnly={!links} type="cpe">
                  {deprecatedBy}
                </DetailsLink>
              </TableData>
            </TableRow>
          )}
          {isDefined(updateTime) && (
            <TableRow>
              <TableData>{_('Last updated')}</TableData>
              <TableData>
                <DateTime date={updateTime} />
              </TableData>
            </TableRow>
          )}
          {isDefined(severity) && (
            <TableRow>
              <TableData>{_('Severity')}</TableData>
              <TableData>
                <SeverityBar severity={severity} />
              </TableData>
            </TableRow>
          )}
        </TableBody>
      </InfoTable>
    </Layout>
  );
};

CpeDetails.propTypes = {
  entity: PropTypes.model.isRequired,
  links: PropTypes.bool,
};

export default CpeDetails;
