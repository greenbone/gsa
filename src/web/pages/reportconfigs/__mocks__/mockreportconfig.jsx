/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

export const mockReportConfig = {
  _id: '12345',
  name: 'foo',
  comment: 'bar',
  creation_time: '2019-07-16T06:31:29Z',
  modification_time: '2019-07-16T06:44:55Z',
  owner: {name: 'admin'},
  writable: '1',
  in_use: '0',
  report_format: {
    _id: '123456',
    name: 'example-configurable-1',
  },
  param: [
    {
      name: 'StringParam',
      value: {
        __text: 'StringValue',
        _using_default: '1',
      },
      default: 'StringValue',
      type: {
        __text: 'string',
        min: '0',
        max: '100',
      },
    },
    {
      name: 'TextParam',
      value: {
        __text: 'TextValue',
        _using_default: '0',
      },
      default: 'TextDefault',
      type: {
        __text: 'text',
        min: '0',
        max: '1000',
      },
    },
    {
      name: 'IntegerParam',
      value: {
        __text: '12',
        _using_default: '1',
      },
      default: '12',
      type: {
        __text: 'integer',
        min: 0,
        max: 50,
      },
    },
    {
      name: 'BooleanParam',
      value: {
        __text: '1',
        _using_default: '0',
      },
      default: '0',
      type: {
        __text: 'boolean',
        min: 0,
        max: 1,
      },
    },
    {
      name: 'SelectionParam',
      value: {
        __text: 'OptionB',
        _using_default: '0',
      },
      default: 'OptionA',
      options: {
        option: ['OptionA', 'OptionB', 'OptionC'],
      },
      type: {
        __text: 'selection',
        min: 0,
        max: 1,
      },
    },
    {
      name: 'ReportFormatListParam',
      value: {
        __text: '654321,7654321',
        report_format: [
          {
            _id: '654321',
            name: 'non-configurable-1',
          },
          {
            _id: '7654321',
            name: 'non-configurable-2',
          },
        ],
        _using_default: '0',
      },
      default: {
        __text: '7654321,1234567',
        report_format: [
          {
            _id: '7654321',
            name: 'non-configurable-2',
          },
          {
            _id: '1234567',
            name: 'example-configurable-2',
          },
        ],
      },
      type: {
        __text: 'report_format_list',
        min: 0,
        max: 1,
      },
    },
  ],
  alerts: {
    alert: [
      {
        _id: '321',
        name: 'ABC',
      },
      {
        _id: '789',
        name: 'XYZ',
      },
    ],
  },
};
