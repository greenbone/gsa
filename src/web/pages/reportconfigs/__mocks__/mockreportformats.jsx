/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */


export const mockReportFormats = [
  {
    id: '123456',
    name: 'example-configurable-1',
    configurable: true,
  },
  {
    id: '654321',
    name: 'non-configurable-1',
    configurable: false,
  },
  {
    id: '1234567',
    name: 'example-configurable-2',
    configurable: true,
  },
  {
    id: '7654321',
    name: 'non-configurable-2',
    configurable: false,
  },
];
