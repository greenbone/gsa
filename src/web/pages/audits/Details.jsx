/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import React from 'react';
import {connect} from 'react-redux';
import {duration} from 'gmp/models/date';
import {scannerTypeName} from 'gmp/models/scanner';
import {isDefined} from 'gmp/utils/identity';
import DateTime from 'web/components/date/DateTime';
import HorizontalSep from 'web/components/layout/HorizontalSep';
import Layout from 'web/components/layout/Layout';
import DetailsLink from 'web/components/link/DetailsLink';
import TableBody from 'web/components/table/Body';
import TableData from 'web/components/table/Data';
import DetailsTable from 'web/components/table/DetailsTable';
import TableRow from 'web/components/table/TableRow';
import DetailsBlock from 'web/entity/Block';
import {compareAlerts} from 'web/pages/tasks/Details';
import {
  loadEntity as loadPolicy,
  selector as policySelector,
} from 'web/store/entities/policies';
import {
  loadEntity as loadSchedule,
  selector as scheduleSelector,
} from 'web/store/entities/schedules';
import compose from 'web/utils/Compose';
import PropTypes from 'web/utils/PropTypes';
import {renderYesNo} from 'web/utils/Render';
import withGmp from 'web/utils/withGmp';
import withTranslation from 'web/utils/withTranslation';

class AuditDetails extends React.Component {
  componentDidMount() {
    const {entity} = this.props;

    if (isDefined(entity.config)) {
      this.props.loadPolicy(entity.config.id);
    }
    if (isDefined(entity.schedule)) {
      this.props.loadSchedule(entity.schedule.id);
    }
  }

  render() {
    const {_} = this.props;

    const {links = true, entity, policy, schedule} = this.props;
    const {
      alerts,
      auto_delete,
      auto_delete_data,
      average_duration,
      config,
      hosts_ordering,
      in_assets,
      last_report,
      scanner,
      schedule_periods,
      target,
      max_checks,
      max_hosts,
    } = entity;

    let dur;
    const hasDuration =
      isDefined(last_report) && isDefined(last_report.scan_start);
    if (hasDuration) {
      if (isDefined(last_report.scan_end)) {
        const diff = last_report.scan_end.diff(last_report.scan_start);
        dur = duration(diff).humanize();
      } else {
        dur = _('Not finished yet');
      }
    } else {
      dur = _('No scans yet');
    }

    const hasAvDuration = isDefined(average_duration) && average_duration > 0;
    const avDuration = hasAvDuration ? average_duration.humanize() : '';

    return (
      <Layout flex="column" grow="1">
        {isDefined(target) && (
          <DetailsBlock title={_('Target')}>
            <DetailsLink id={target.id} textOnly={!links} type="target">
              {target.name}
            </DetailsLink>
          </DetailsBlock>
        )}

        {isDefined(alerts) && alerts.length > 0 && (
          <DetailsBlock title={_('Alerts')}>
            <HorizontalSep>
              {alerts.sort(compareAlerts).map(alert => (
                <span key={alert.id}>
                  <DetailsLink id={alert.id} textOnly={!links} type="alert">
                    {alert.name}
                  </DetailsLink>
                </span>
              ))}
            </HorizontalSep>
          </DetailsBlock>
        )}

        {isDefined(scanner) && (
          <DetailsBlock title={_('Scanner')}>
            <DetailsTable>
              <TableBody>
                <TableRow>
                  <TableData>{_('Name')}</TableData>
                  <TableData>
                    <span>
                      <DetailsLink
                        id={scanner.id}
                        textOnly={!links}
                        type="scanner"
                      >
                        {scanner.name}
                      </DetailsLink>
                    </span>
                  </TableData>
                </TableRow>
                <TableRow>
                  <TableData>{_('Type')}</TableData>
                  <TableData>{scannerTypeName(scanner.scannerType)}</TableData>
                </TableRow>
                {isDefined(config) && (
                  <TableRow>
                    <TableData>{_('Policy')}</TableData>
                    <TableData>
                      <span>
                        <DetailsLink
                          id={config.id}
                          textOnly={!links}
                          type="policy"
                        >
                          {config.name}
                        </DetailsLink>
                      </span>
                    </TableData>
                  </TableRow>
                )}
                {isDefined(policy) && (
                  <TableRow>
                    <TableData>{_('Order for target hosts')}</TableData>
                    <TableData>{hosts_ordering}</TableData>
                  </TableRow>
                )}
                {isDefined(policy) && isDefined(max_checks) && (
                  <TableRow>
                    <TableData>
                      {_('Maximum concurrently executed NVTs per host')}
                    </TableData>
                    <TableData>{max_checks}</TableData>
                  </TableRow>
                )}
                {isDefined(policy) && isDefined(max_hosts) && (
                  <TableRow>
                    <TableData>
                      {_('Maximum concurrently scanned hosts')}
                    </TableData>
                    <TableData>{max_hosts}</TableData>
                  </TableRow>
                )}
              </TableBody>
            </DetailsTable>
          </DetailsBlock>
        )}

        <DetailsBlock title={_('Assets')}>
          <DetailsTable>
            <TableBody>
              <TableRow>
                <TableData>{_('Add to Assets')}</TableData>
                <TableData>{renderYesNo(in_assets)}</TableData>
              </TableRow>
            </TableBody>
          </DetailsTable>
        </DetailsBlock>

        {isDefined(schedule) && (
          <DetailsBlock title={_('Schedule')}>
            <DetailsTable>
              <TableBody>
                <TableRow>
                  <TableData>{_('Name')}</TableData>
                  <TableData>
                    <span>
                      <DetailsLink
                        id={schedule.id}
                        textOnly={!links}
                        type="schedule"
                      >
                        {schedule.name}
                      </DetailsLink>
                    </span>
                  </TableData>
                </TableRow>
                {isDefined(schedule.event) && (
                  <TableRow>
                    <TableData>{_('Next')}</TableData>
                    <TableData>
                      {isDefined(schedule.event.nextDate) ? (
                        <DateTime date={schedule.event.nextDate} />
                      ) : (
                        _('N/A')
                      )}
                    </TableData>
                  </TableRow>
                )}
              </TableBody>
            </DetailsTable>
          </DetailsBlock>
        )}

        <DetailsBlock title={_('Scan')}>
          <DetailsTable>
            <TableBody>
              <TableRow>
                <TableData>{_('Duration of last Scan')}</TableData>
                <TableData>{dur}</TableData>
              </TableRow>
              {hasAvDuration && (
                <TableRow>
                  <TableData>{_('Average Scan duration')}</TableData>
                  <TableData>{avDuration}</TableData>
                </TableRow>
              )}
              {schedule_periods > 0 && (
                <TableRow>
                  <TableData>{_('Period')}</TableData>
                  <TableData>
                    {schedule_periods > 1
                      ? _('{{nr}} more times', {nr: schedule_periods})
                      : _('Once')}
                  </TableData>
                </TableRow>
              )}
              <TableRow>
                <TableData>{_('Auto delete Reports')}</TableData>
                <TableData>
                  {auto_delete === 'keep'
                    ? _(
                        'Automatically delete oldest reports but always keep ' +
                          'newest {{nr}} reports',
                        {nr: auto_delete_data},
                      )
                    : _('Do not automatically delete reports')}
                </TableData>
              </TableRow>
            </TableBody>
          </DetailsTable>
        </DetailsBlock>
      </Layout>
    );
  }
}

AuditDetails.propTypes = {
  entity: PropTypes.model.isRequired,
  gmp: PropTypes.gmp.isRequired,
  links: PropTypes.bool,
  loadPolicy: PropTypes.func.isRequired,
  loadSchedule: PropTypes.func.isRequired,
  policy: PropTypes.model,
  schedule: PropTypes.model,
  _: PropTypes.func.isRequired,
};

const mapStateToProps = (rootState, {entity = {}}) => {
  const scheduleSel = scheduleSelector(rootState);
  const policySel = policySelector(rootState);
  return {
    policy: isDefined(entity.config)
      ? policySel.getEntity(entity.config.id)
      : undefined,
    schedule: isDefined(entity.schedule)
      ? scheduleSel.getEntity(entity.schedule.id)
      : undefined,
  };
};

const mapDispatchToProps = (dispatch, {gmp}) => ({
  loadPolicy: id => dispatch(loadPolicy(gmp)(id)),
  loadSchedule: id => dispatch(loadSchedule(gmp)(id)),
});

export default compose(
  withTranslation,
  withGmp,
  connect(mapStateToProps, mapDispatchToProps),
)(AuditDetails);
