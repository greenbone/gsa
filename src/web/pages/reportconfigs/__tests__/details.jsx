/* Copyright (C) 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';

import Capabilities from 'gmp/capabilities/capabilities';
import ReportConfig from 'gmp/models/reportconfig';

import {rendererWith} from 'web/utils/testing';

import Details from '../details';
import {mockReportConfig} from '../__mocks__/mockreportconfig';

describe('Report Config Details tests', () => {
  test('should render full Details', () => {
    const config = ReportConfig.fromElement(mockReportConfig);

    const caps = new Capabilities(['everything']);

    const {render} = rendererWith({capabilities: caps, router: true});

    const {element, getAllByTestId} = render(<Details entity={config} />);

    expect(element).toMatchSnapshot();

    expect(element).toHaveTextContent('StringParam');
    expect(element).toHaveTextContent('StringValue');

    expect(element).toHaveTextContent('TextParam');
    const preElements = element.querySelectorAll('pre');
    expect(preElements[0]).toHaveTextContent('TextValue');

    expect(element).toHaveTextContent('IntegerParam');
    expect(element).toHaveTextContent('12');
    expect(element).toHaveTextContent('SelectionParam');
    expect(element).toHaveTextContent('OptionB');
    expect(element).toHaveTextContent('ReportFormatListParam');
    expect(element).toHaveTextContent('non-configurable');

    const detailslinks = getAllByTestId('details-link');

    // Report format of the config
    expect(detailslinks[0]).toHaveTextContent('example-configurable-1');
    expect(detailslinks[0]).toHaveAttribute('href', '/reportformat/123456');

    // Report format params
    expect(detailslinks[1]).toHaveTextContent('non-configurable-1');
    expect(detailslinks[1]).toHaveAttribute('href', '/reportformat/654321');

    expect(detailslinks[2]).toHaveTextContent('non-configurable-2');
    expect(detailslinks[2]).toHaveAttribute('href', '/reportformat/7654321');

    // Alerts
    expect(detailslinks[3]).toHaveTextContent('ABC');
    expect(detailslinks[3]).toHaveAttribute('href', '/alert/321');

    expect(detailslinks[4]).toHaveTextContent('XYZ');
    expect(detailslinks[4]).toHaveAttribute('href', '/alert/789');
  });

  test('should render orphaned config details', () => {
    const config = ReportConfig.fromElement({
      _id: '123',
      name: 'foo',
      comment: 'bar',
      orphan: '1',
    });

    const caps = new Capabilities(['everything']);

    const {render} = rendererWith({capabilities: caps, router: true});

    const {element} = render(<Details entity={config} />);

    expect(element).toMatchSnapshot();

    expect(element).toHaveTextContent(
      'not available for orphaned report configs',
    );
  });
});
