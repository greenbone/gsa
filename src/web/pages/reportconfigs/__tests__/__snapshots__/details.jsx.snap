// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Report Config Details tests > should render full Details 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  border: 0;
  border-spacing: 0px;
  font-size: 12px;
  text-align: left;
  table-layout: auto;
}

.c7 {
  vertical-align: top;
}

.c2 td {
  padding: 4px 4px 4px 0;
}

.c2 tr td:first-child {
  padding-right: 5px;
}

.c3 {
  width: 10%;
}

.c4 {
  width: 90%;
}

@media print {
  .c1 {
    border-collapse: collapse;
  }
}

<div
  class="c0"
>
  <table
    class="c1 c2"
  >
    <colgroup>
      <col
        class="c3"
        width="10%"
      />
      <col
        class="c4"
        width="90%"
      />
    </colgroup>
    <tbody
      class=""
    >
      <tr>
        <td>
          <div
            class="c5"
          >
            Report Format
          </div>
        </td>
        <td>
          <div
            class="c5"
          >
            <span>
              <a
                class="c6"
                data-testid="details-link"
                href="/reportformat/123456"
              >
                example-configurable-1
              </a>
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="c7"
        >
          <div
            class="c5"
          >
            Parameters
          </div>
        </td>
        <td>
          <div
            class="c5"
          >
            <div
              class="c8"
            >
              <table
                class="c1 c2"
              >
                <tbody
                  class=""
                >
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        StringParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        StringValue
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        TextParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        <pre>
                          TextValue
                        </pre>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        IntegerParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        12
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        BooleanParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        Yes
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        SelectionParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        OptionB
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        ReportFormatListParam
                      </div>
                    </td>
                    <td>
                      <div
                        class="c5"
                      >
                        <a
                          class="c6"
                          data-testid="details-link"
                          href="/reportformat/654321"
                        >
                          non-configurable-1
                        </a>
                        <a
                          class="c6"
                          data-testid="details-link"
                          href="/reportformat/7654321"
                        >
                          non-configurable-2
                        </a>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="c7"
        >
          <div
            class="c5"
          >
            Alerts using this Report Config
          </div>
        </td>
        <td>
          <div
            class="c5"
          >
            <span>
              <a
                class="c6"
                data-testid="details-link"
                href="/alert/321"
              >
                ABC
              </a>
            </span>
            <span>
              <a
                class="c6"
                data-testid="details-link"
                href="/alert/789"
              >
                XYZ
              </a>
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Report Config Details tests > should render orphaned config details 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 0;
  border-spacing: 0px;
  font-size: 12px;
  text-align: left;
  table-layout: auto;
}

.c6 {
  vertical-align: top;
}

.c2 td {
  padding: 4px 4px 4px 0;
}

.c2 tr td:first-child {
  padding-right: 5px;
}

.c3 {
  width: 10%;
}

.c4 {
  width: 90%;
}

@media print {
  .c1 {
    border-collapse: collapse;
  }
}

<div
  class="c0"
>
  <table
    class="c1 c2"
  >
    <colgroup>
      <col
        class="c3"
        width="10%"
      />
      <col
        class="c4"
        width="90%"
      />
    </colgroup>
    <tbody
      class=""
    >
      <tr>
        <td>
          <div
            class="c5"
          >
            Report Format
          </div>
        </td>
        <td>
          <div
            class="c5"
          >
            <span />
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="c6"
        >
          <div
            class="c5"
          >
            Parameters
          </div>
        </td>
        <td>
          <div
            class="c5"
          >
            <div
              class="c7"
            >
              <table
                class="c1 c2"
              >
                <tbody
                  class=""
                >
                  <tr>
                    <td>
                      <div
                        class="c5"
                      >
                        <i>
                          not available for orphaned report configs
                        </i>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
